### PR TITLE
Add the text area to the MarkdownPreview Component

### PR DESCRIPTION
**Pull request by dmcavoy**
A while back in https://github.com/code-dot-org/code-dot-org/pull/36367 @uponthesun suggested pulling the text area in with the preview into one component.  This does that work to create one component which is a text area and a preview.  I updated the place on the script and course edit pages that were using the previous MarkdownPreview Component.

Screen Shot 2020-09-08 at 3 58 53 PM

## Testing story

Updated the tests accordingly.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/course-editor/CourseEditor.js [hunk 1]
```diff
@@ -134,28 +134,20 @@ export default class CourseEditor extends Component {
...
-            name="description_teacher"
-            defaultValue={this.state.descriptionTeacher}
-            rows={5}
-            style={styles.input}
-            onChange={this.handleTeacherDescriptionChange}
-          />
-          <MarkdownPreview markdown={this.state.descriptionTeacher} />
-        </label>
+        <TextareaWithMarkdownPreview
+          markdown={this.state.descriptionStudent}
```
Comment (uponthesun) [inline apps/src/lib/levelbuilder/course-editor/CourseEditor.js]: Should the `TextareaWithMarkdownPreview` component also own the markdown state and the onChange handler?

Comment (dmcavoy) [reply apps/src/lib/levelbuilder/course-editor/CourseEditor.js]: Great idea! I have updated with that change

File: apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx [hunk 2]
```diff
@@ -172,17 +167,12 @@ export default class ScriptEditor extends React.Component {
...
-            name="description"
-            defaultValue={this.state.description}
-            rows={5}
-            style={styles.input}
-            onChange={this.handleDescriptionChange}
-          />
-          <MarkdownPreview markdown={this.state.description} />
-        </label>
+        <TextareaWithMarkdownPreview
+          markdown={this.state.description}
```
Comment (uponthesun) [inline apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx]: `this.props.i18nData.description`?

**Review by uponthesun — APPROVED** [review]

File: apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx (hunk 2 above)
Comment (dmcavoy) [reply apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx]: good catch!

**Pull request merged**